### PR TITLE
간단한 socket 예제 작성

### PR DESCRIPTION
**Pull request by ors-woon**
서버/ 클라이언트 작성

서버 - 연결 후 끊김 

클라이언트 - 연결 후 출력 

뼈대 작성.

### CODE REVIEW

File: src/main/java/server/Server.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import java.net.Socket;
+
+
+public class Server {
+
+    private static final int PORT = 8080;
+
+    public static void main(String[] args) {
+        try {
+            ServerSocket serverSocket = new ServerSocket(PORT);
```
Comment (ChulWoonJang) [inline src/main/java/server/Server.java]: Bug Bug: Use try-with-resources or close this "ServerSocket" in a "finally" clause. ([squid:S2095](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS2095))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3H7_bEzQ-kUL94F&open=AWQTV3H7_bEzQ-kUL94F)

[ISSUE_KEY]: AWQTV3H7_bEzQ-kUL94F

File: src/main/java/client/Client.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+public class Client {
+
+    private static final int PORT = 8080;
+    private static final String HOST = "127.0.0.1";
+
+
+    public static void main(String[] args) {
+        try {
+            Socket socket = new Socket(HOST, PORT);
```
Comment (ChulWoonJang) [inline src/main/java/client/Client.java]: Bug Bug: Use try-with-resources or close this "Socket" in a "finally" clause. ([squid:S2095](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS2095))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3ID_bEzQ-kUL94J&open=AWQTV3ID_bEzQ-kUL94J)

[ISSUE_KEY]: AWQTV3ID_bEzQ-kUL94J

File: src/main/java/client/Client.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            BufferedReader br = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            String str = null;
+            while ((str = br.readLine()) != null ){
+                System.out.println(str);
```
Comment (ChulWoonJang) [inline src/main/java/client/Client.java]: Code Smell Code Smell: Replace this use of System.out or System.err by a logger. ([squid:S106](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS106))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3ID_bEzQ-kUL94L&open=AWQTV3ID_bEzQ-kUL94L)

[ISSUE_KEY]: AWQTV3ID_bEzQ-kUL94L

File: src/main/java/client/Client.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            }
+
+            socket.close();
+
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (ChulWoonJang) [inline src/main/java/client/Client.java]: Vulnerability Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3ID_bEzQ-kUL94N&open=AWQTV3ID_bEzQ-kUL94N)

[ISSUE_KEY]: AWQTV3ID_bEzQ-kUL94N

File: src/main/java/client/Client.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package main.java.client;
+
+import java.io.*;
+import java.net.ServerSocket;
```
Comment (ChulWoonJang) [inline src/main/java/client/Client.java]: Code Smell Code Smell: Remove this unused import 'java.net.ServerSocket'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3ID_bEzQ-kUL94M&open=AWQTV3ID_bEzQ-kUL94M)

[ISSUE_KEY]: AWQTV3ID_bEzQ-kUL94M

File: src/main/java/server/Server.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+                OutputStream stream = socket.getOutputStream();
+                stream.write("helloWorld".getBytes());
+            } catch (Exception e) {
+                e.printStackTrace();
+            } finally {
+                socket.close();
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (ChulWoonJang) [inline src/main/java/server/Server.java]: Vulnerability Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3H7_bEzQ-kUL94I&open=AWQTV3H7_bEzQ-kUL94I)

[ISSUE_KEY]: AWQTV3H7_bEzQ-kUL94I

**Review by ChulWoonJang — COMMENTED** [review]
SonarQube analysis found issues:
Bug Bugs: 2
Vulnerability Vulnerabilities: 4
Code Smell Code Smells: 3

[See all issues in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&resolved=false)

File: src/main/java/server/Server.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+    public static void main(String[] args) {
+        try {
+            ServerSocket serverSocket = new ServerSocket(PORT);
+            Socket socket = serverSocket.accept(); // block
+            try {
+                OutputStream stream = socket.getOutputStream();
+                stream.write("helloWorld".getBytes());
+            } catch (Exception e) {
+                e.printStackTrace();
```
Comment (ChulWoonJang) [inline src/main/java/server/Server.java]: Vulnerability Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3H7_bEzQ-kUL94H&open=AWQTV3H7_bEzQ-kUL94H)

[ISSUE_KEY]: AWQTV3H7_bEzQ-kUL94H

File: src/main/java/client/Client.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import java.net.Socket;
+
+
+public class Client {
+
+    private static final int PORT = 8080;
+    private static final String HOST = "127.0.0.1";
```
Comment (ChulWoonJang) [inline src/main/java/client/Client.java]: Vulnerability Vulnerability: Make this IP "127.0.0.1" address configurable. ([squid:S1313](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS1313))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3ID_bEzQ-kUL94K&open=AWQTV3ID_bEzQ-kUL94K)

[ISSUE_KEY]: AWQTV3ID_bEzQ-kUL94K

File: src/main/java/server/Server.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+public class Server {
+
+    private static final int PORT = 8080;
+
+    public static void main(String[] args) {
+        try {
+            ServerSocket serverSocket = new ServerSocket(PORT);
+            Socket socket = serverSocket.accept(); // block
+            try {
```
Comment (ChulWoonJang) [inline src/main/java/server/Server.java]: Code Smell Code Smell: Extract this nested try block into a separate method. ([squid:S1141](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS1141))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3H7_bEzQ-kUL94G&open=AWQTV3H7_bEzQ-kUL94G)

[ISSUE_KEY]: AWQTV3H7_bEzQ-kUL94G

File: src/main/java/client/Client.java (hunk 8 above)
Comment (JangChulwoon) [reply src/main/java/client/Client.java]: ip는 하드코딩을 하지 말라는 이슈.
이 부분은 나중에 파일로 빼든가 해야겠다 .
(당장은 건들기가 귀찮은데..)

**Comment by JangChulwoon** [comment]
버전 에러가 뜨는데, 이유 좀더 보고 .. 고치자
졸립다 ..

**Review by ChulWoonJang — COMMENTED** [review]
SonarQube analysis found issues:
Bug Bugs: 0
Vulnerability Vulnerabilities: 3
Code Smell Code Smells: 1

[See all issues in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&resolved=false)

File: src/main/java/client/Client.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    public static void main(String[] args) {
+        try(Socket socket = new Socket(HOST, PORT)) {
+            BufferedReader br = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            String str;
+            while ((str = br.readLine()) != null ){
+                System.out.println(str);
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (ChulWoonJang) [inline src/main/java/client/Client.java]: Vulnerability Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3ID_bEzQ-kUL94N&open=AWQTV3ID_bEzQ-kUL94N)

[ISSUE_KEY]: AWQTV3ID_bEzQ-kUL94N

File: src/main/java/server/Server.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+public class Server {
+
+    private static final int PORT = 8080;
+
+    public static void main(String[] args) {
+        try (Socket socket  = new ServerSocket(PORT).accept()) {
+            OutputStream stream = socket.getOutputStream();
+            stream.write("helloWorld".getBytes());
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (ChulWoonJang) [inline src/main/java/server/Server.java]: Vulnerability Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3H7_bEzQ-kUL94I&open=AWQTV3H7_bEzQ-kUL94I)

[ISSUE_KEY]: AWQTV3H7_bEzQ-kUL94I

File: src/main/java/client/Client.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
+package main.java.client;
+
+import java.io.*;
+import java.net.Socket;
+
+public class Client {
+
+    private static final int PORT = 8080;
+    private static final String HOST = "127.0.0.1";
```
Comment (ChulWoonJang) [inline src/main/java/client/Client.java]: Vulnerability Vulnerability: Make this IP "127.0.0.1" address configurable. ([squid:S1313](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS1313))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3ID_bEzQ-kUL94K&open=AWQTV3ID_bEzQ-kUL94K)

[ISSUE_KEY]: AWQTV3ID_bEzQ-kUL94K

File: src/main/java/client/Client.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+
+    public static void main(String[] args) {
+        try(Socket socket = new Socket(HOST, PORT)) {
+            BufferedReader br = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            String str;
+            while ((str = br.readLine()) != null ){
+                System.out.println(str);
```
Comment (ChulWoonJang) [inline src/main/java/client/Client.java]: Code Smell Code Smell: Replace this use of System.out or System.err by a logger. ([squid:S106](https://sonarcloud.io/organizations/jangchulwoon-github/rules#rule_key=squid%3AS106))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=groupId%3AHTTPServer&pullRequest=3&issues=AWQTV3ID_bEzQ-kUL94L&open=AWQTV3ID_bEzQ-kUL94L)

[ISSUE_KEY]: AWQTV3ID_bEzQ-kUL94L

**Pull request merged**